apps/web/src/components/invoice-generator/index.tsx

```diff
@@ -308,11 +309,38 @@ function InvoiceGeneratorForm() {
 
     event.preventDefault();
     const formData = new FormData(event.target);
+
+    const payloadSize = calculateFormDataSize(formData);
+    
+    if(payloadSize > MAX_PAYLOAD_SIZE){
+      toast.error("Total form size exceeds 24MB limit. Please remove some files.");
+      return;
+    }
+
     startTransition(() => {
       formAction(formData);
     });
   };
 
+  const calculateFormDataSize = (formData: FormData):number => {
+    let totalSize = 0;
+
+    formData.forEach((value, key) => {
+    // Add size of the key
+      totalSize += new Blob([key]).size;
+
+      if (value instanceof File) {
+        // Add file size
+        totalSize += value.size;
+      } else {
+        // Add string value size
+        totalSize += new Blob([value.toString()]).size;
+      }
+    });
```

Comment:
Nitpick: I would remove these comments that don't add any value

```suggestion
    formData.forEach((value, key) => {
      totalSize += new Blob([key]).size;

      if (value instanceof File) {
        totalSize += value.size;
      } else {
        totalSize += new Blob([value.toString()]).size;
      }
    });
```